### PR TITLE
(8) Put gnucash-docs.spec into tarball to build RPM file properly.

### PR DESCRIPTION
**Pull request by usamimikamen-sk**
Depens on #221
---
The file gnucash-docs.spec.in exists at gnucash-docs/ but it isn't configured and included into gnucash-docs.tar.gz.
This PR fixes this problem.

After this PR is accepted, the following commands will work fine.

* `rpmbuild -ta gnucash-docs-${version}.tar.gz`
* `LANG=de_DE yelp help:gnucash-help`

### CODE REVIEW

**Comment by jralls** [comment]
You seem to be branching your feature branches from other feature branches instead of on git maint and that's causing extra commits to get pulled in, making it a bit confusing about where this PR starts; it also means that the PRs need to be merged in the right order to avoid conflicts. Is there a way to sort this out without too much work?

**Comment by gjanssens** [comment]
I think rather than fixing the rpm spec file it's better to drop it. Here's my reasoning:
1. An rpm is a package format, much like the Windows installer, the Macos dmg or flatpak. We have separate repos to handle packaging.
2. I strongly doubt there are any users building rpms based on the spec files we provide. Particularly because the nitty gritty details for the spec file are typically  distro dependent. What we provide hasn't been maintained for years and is unlikely to be on par with current best practises for rpm packaging.
3. Each distro that packages gnucash as rpm maintains its own distro-specific variant of an rpm spec file. They have their own repositories to manage them. These variants are in much better shape than our spec file. Should a user want to build their own gnucash rpm, they should really use the spec file used by their distro, not ours.

**Comment by fellen** [comment]
Then move them to gnucash[-docs]-on-rpm, but a repo for one file?

Historically there has been some exchange between our and at least opensuse. but other rpm based distros will benefit, too.

I prefer to maintain one file instead of a list of maintainers to announce changes.

**Comment by usamimikamen-sk** [comment]
> You seem to be branching your feature branches from other feature branches instead of on git maint and that's causing extra commits to get pulled in, making it a bit confusing about where this PR starts; it also means that the PRs need to be merged in the right order to avoid conflicts. Is there a way to sort this out without too much work?

There is no easy way because these changesets strongly depend on  PR #211, #213, and #217.
And the order is very important.

For example, PR #217 "Make DTD ENTITY(s) translatable" depends on #211. The order is (4.5). But it was merged into maint before #211. Therefore @CWehli and @gjanssens had an accident that DTD files were not installed into the appropriate location.
Their problem will be gone after PR #211 is merged.

PR #214 is cherry-picked, but the contents of README include changes in #211. Therefore README in the current maint doesn't match the source tree.

If a dependency is not important,  I will create PR like #222.

**Comment by usamimikamen-sk** [comment]
> You seem to be branching your feature branches from other feature branches

Well, this is one large branch to clean up the build system. @jralls and @fellen claimed me to separate PR in #217 and #204, respectively. So I created small PRs.

I wish GitHub had PR accumulation function.

**Comment by jralls** [comment]
It's not github's fault, it's just how git works. Maybe edit the head comment on each one to list the prerequisite PRs that have to be merged before that one can be?

File: gnucash-docs.spec.in [hunk 1]
```diff
@@ -26,24 +24,42 @@ This is the documentation module for GnuCash.
...
-%dir %{_datadir}/gnome/help/gnucash-help
-%doc %{_datadir}/gnome/help/gnucash-help/C/
+%cmake_install
+
+%files
+%license COPYING-DOCS
+%doc COPYING COPYING-DOCS AUTHORS HACKING NEWS README
+%dir %{_datadir}/help
+
+%dir %{_datadir}/help/C/gnucash-guide
```
Comment (usamimikamen-sk) [inline gnucash-docs.spec.in:38]: Use %{find_lang} if available.

**Comment by usamimikamen-sk** [comment]
> It's not github's fault, it's just how git works. Maybe edit the head comment on each one to list the prerequisite PRs that have to be merged before that one can be?

I've already changed the title. (1)-(7) should be merged in order.

**Comment by usamimikamen-sk** [comment]
I wrote the dependency at the top of the comments in PRs.

**Comment by gjanssens** [comment]
> Then move them to gnucash[-docs]-on-rpm, but a repo for one file?
> 
No just plain old removal. Period.

> Historically there has been some exchange between our and at least opensuse. but other rpm based distros will benefit, too.

Eh, I see you chose to merge the opensuse spec last year in June. Before that the last commit was in 2007, a trivial change by Derek.

That just illustrates my point: we don't maintain the spec file and we  don't use it. Each rpm based distro by now has its own spec and is no longer interested in ours. It's only cruft we keep carrying around.
> 
> I prefer to maintain one file instead of a list of maintainers to announce changes.

This argument makes no sense. The spec has nothing to do with announcing changes to maintainers. Maintainers follow our release announcements to keep up.

**Comment by usamimikamen-sk** [comment]
I modified the RPM spec file because someone will create RPM to check using yelp.
But nobody seems to use this file. I want to remove it.

**Comment by fellen** [comment]
My reason to update the rpm file was that opensuse was years behind after our SF download directory changed. 

There are more "packaging files": https://github.com/Gnucash/gnucash-docs/tree/maint/.github/workflows
Should we remove them, too?

**Comment by jralls** [comment]
> My reason to update the rpm file was that opensuse was years behind after our SF download directory changed.

That's the OpenSuSE maintainer's problem and it doesn't make sense for us to maintain a gnucash-docs.spec for one distro out of 100. I agree with @gjanssens, it's out of scope for GnuCash to maintain distro files. We don't have the information needed to maintain them properly.

> There are more "packaging files": https://github.com/Gnucash/gnucash-docs/tree/maint/.github/workflows
Should we remove them, too?

That's incorrect. Those workflows test building from the commit that initiates the CI. They don't do anything about packaging gnucash-docs beyond making the source tarball.

**Pull request closed without merging**